src/include/duckdb/storage/statistics/base_statistics.hpp

```diff
@@ -23,39 +23,55 @@ class FieldWriter;
 class FieldReader;
 class Vector;
 class ValidityStatistics;
+class DistinctStatistics;
+struct VectorData;
 
 class BaseStatistics {
 public:
-	explicit BaseStatistics(LogicalType type);
+	BaseStatistics(LogicalType type, bool global);
```

Comment:
Perhaps we could turn this into a more descriptive enum instead of a bool, e.g. `StatisticsType::GLOBAL_STATS`, `StatisticsType::LOCAL_STATS`?